### PR TITLE
Bulk allocation of funds and ability to set (instead of add) balances

### PR DESCRIPTION
**Pull request by gustavlrsn**
Adds ability on the event members page to bulk allocate funds to members, both adding and setting. Also adds the Set option for individual allocations.

### CODE REVIEW

File: ui/components/members/AllocateModal.js [hunk 1]
```diff
@@ -34,14 +40,20 @@ const AllocateModal = ({ member, event, handleClose }) => {
...
+          selected={type}
+          className="mx-auto mb-4"
+        />
+        {/* <p className="text-center mt-4 mb-2">
           <span>
             {thousandSeparator(member.balance / 100)} {event.currency}{" "}
           </span>{" "}
           +
-        </p>
+        </p> */}
```
Comment (Powersource) [inline ui/components/members/AllocateModal.js:56]: remove

File: ui/components/members/AllocateModal.js [hunk 2]
```diff
@@ -61,9 +73,23 @@ const AllocateModal = ({ member, event, handleClose }) => {
             autoFocus
             className="w-36 mx-auto mb-2"
           />
-          <p className="text-center mb-4">
-            = {thousandSeparator(total / 100)} {event.currency} total
+          <p className="text-center mb-4 text-sm text-gray-800">
+            {type === "Add" ? (
+              <>
+                Adding {thousandSeparator(amount / 100)} {event.currency} to{" "}
```
Comment (Powersource) [inline ui/components/members/AllocateModal.js:79]: Maybe rename `amount` to cents or something? Idk not super needed but I started mixing up all the different variable names when I got to this part :p

File: ui/components/members/BulkAllocateModal.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import { useState } from "react";
```
Comment (Powersource) [inline ui/components/members/BulkAllocateModal.js:1]: Could maybe be nice to refactor this file and AllocateModal to deduplicate code in some way. But it's not a ton of duplication so only if you feel like it

File: api/resolvers/index.js [hunk 4]
```diff
@@ -1651,15 +1651,116 @@ const resolvers = {
...
+
+        const adjustment = amount - balance;
+
+        await new Allocation({
+          organizationId: currentOrgMember.organizationId,
+          eventId: targetEventMember.eventId,
+          eventMemberId,
+          amount: adjustment,
+        }).save();
+      }
```
Comment (Powersource) [inline api/resolvers/index.js:1694]: Could be nice with an `} else {` to catch if you send a bad type (or does the graphql enum maybe check that for you?)

File: api/resolvers/index.js [hunk 5]
```diff
@@ -1651,15 +1651,116 @@ const resolvers = {
...
+
+          const adjustment = amount - balance;
+
+          await new Allocation({
+            organizationId: currentOrgMember.organizationId,
+            eventId,
+            eventMemberId: member.id,
+            amount: adjustment,
+          }).save();
+        }
```
Comment (Powersource) [inline api/resolvers/index.js:1759]: This chunk of code inside the for loop looks like it's mostly the same as the code in the other `allocate` resolver. Would be nice to extract the common logic into a shared function.

File: api/resolvers/index.js (hunk 4 above)
Comment (gustavlrsn) [reply api/resolvers/index.js:1694]: Yep, the graphql enum will take care of that

File: ui/components/members/AllocateModal.js (hunk 1 above)
Comment (gustavlrsn) [reply ui/components/members/AllocateModal.js]: fixed

File: ui/components/members/BulkAllocateModal.js (hunk 3 above)
Comment (gustavlrsn) [reply ui/components/members/BulkAllocateModal.js:1]: Let's go for rule of three, "Three strikes and you refactor" :-)

File: api/resolvers/index.js (hunk 5 above)
Comment (gustavlrsn) [reply api/resolvers/index.js]: I consolidated into a function that I put i `api/controller/index.js` where we can add more controller type functionality, inspired by #75

File: ui/components/members/AllocateModal.js (hunk 2 above)
Comment (gustavlrsn) [reply ui/components/members/AllocateModal.js:73]: All amounts are in cents so renaming to that might just make it more confusing 🤔

Comment (Powersource) [reply ui/components/members/AllocateModal.js:73]: Yeah good point. Maybe just `amount` that's in full units

File: api/resolvers/index.js (hunk 5 above)
Comment (Powersource) [reply api/resolvers/index.js]: Beautiful!

**Pull request merged**